File: Section Reports/Preview/Hyperlinks and DrillThrough/VB.NET/Hyperlinks and DrillThrough/DrillThrough1.vb
Public Class DrillThrough1
	Inherits GrapeCity.ActiveReports.SectionReport
	Public Sub New()
		MyBase.New()
		InitializeComponent()
		'Connect to Sample Database
		'
		DataSource = New Data.OleDBDataSource()
		CType(DataSource, Data.OleDBDataSource).ConnectionString = My.Resources.ConnectionString
		
		CType(DataSource, Data.OleDBDataSource).SQL = "Select * from orders where customerID = '<%param:customerID%>' order by orderdate"
		Dim paramCustomerID As New SectionReportModel.Parameter()
		paramCustomerID.Key = "customerID"
	End Sub
	Private ds As Data.OleDBDataSource
	Private Sub Detail_BeforePrint(ByVal sender As System.Object, ByVal e As EventArgs) Handles Detail.BeforePrint
		txtOrderID.HyperLink = "DrillThrough2:" & txtOrderID.Text
	End Sub
End Class

File: Section Reports/Preview/Hyperlinks and DrillThrough/C#/Hyperlinks and DrillThrough/DrillThrough1.cs
namespace GrapeCity.ActiveReports.Samples.HyperlinksAndDrillThrough
{
	public partial class DrillThrough1 : GrapeCity.ActiveReports.SectionReport
	{
		public DrillThrough1()
		{
			InitializeComponent();
			//Connect to Sample Database
			Data.OleDBDataSource dS = new Data.OleDBDataSource();
			dS.ConnectionString = Properties.Resources.ConnectionString;
			dS.SQL = "Select * from orders where customerID = '<%customerID%>' order by orderdate";
			DataSource = dS;
		}
		/// <summary>
		/// BeforePrint Event
		/// ActiveReports raises this event before the section is rendered to a page.
		/// Use this event to modify the control's values before printing.
		/// The changes made here do not effect the height of the section.
		/// </summary>
		private void Detail_BeforePrint(object sender, System.EventArgs eArgs)
		{
			//Assign the hyperlink value to the Text value
			//
			txtOrderID.HyperLink = "DrillThrough2:" + txtOrderID.Text;
		}
	}
}
